src/jdk.hotspot.agent/macosx/native/libsaproc/libproc_impl.h

```diff
@@ -71,6 +72,45 @@ typedef struct reg {
   register_t      r_rsp;
   register_t      r_ss;          // not used
 } reg;
+#elif defined(__arm64__)
+/*** registers copied from bsd/arm64 */
+typedef struct reg {
+   register_t     r_r0;
+   register_t     r_r1;
+   register_t     r_r2;
+   register_t     r_r3;
+   register_t     r_r4;
+   register_t     r_r5;
+   register_t     r_r6;
+   register_t     r_r7;
+   register_t     r_r8;
+   register_t     r_r9;
+   register_t     r_r10;
+   register_t     r_r11;
+   register_t     r_r12;
+   register_t     r_r13;
+   register_t     r_r14;
+   register_t     r_r15;
+   register_t     r_r16;
+   register_t     r_r17;
+   register_t     r_r18;
+   register_t     r_r19;
+   register_t     r_r20;
+   register_t     r_r21;
+   register_t     r_r22;
+   register_t     r_r23;
+   register_t     r_r24;
+   register_t     r_r25;
+   register_t     r_r26;
+   register_t     r_r27;
+   register_t     r_r28;
+   register_t     r_fp;
+   register_t     r_lr;
+   register_t     r_sp;
+   register_t     r_pc;
+   uint32_t       r_pstate;
+} reg;
```

Comment:
I would leave r18 but remove _lr and _pstate here, as they seem to be unused in sa-agent and can be safely removed

Reply:
I've removed pstate but kept lr.